src/mon/MonMap.cc

```diff
@@ -778,9 +787,14 @@ seastar::future<> MonMap::init_with_dns_srv(bool for_mkfs, const std::string& na
                             record.priority,
                             record.weight,
                             false);
+      }).handle_exception_type([t=record.target](const std::system_error& e) {
+        logger().debug("{}: unable to resolve name for {}: {}",
+                       "init_with_dns_srv", t, e);
```

Comment:
if the inner exception is caught by `handle_exception_type()`, i don't think the outer `.handle_exception_type()` would be able to see it. i think a better fix would be log and ignore the error in the `parallel_for_each()` loop, and drop the outer `handle_exception_type()` call.

what do you think?

Reply:
ahh, you are right! thank you for the explanations.